pkg/service/sip.go

```diff
@@ -390,3 +391,64 @@ func (s *SIPService) CreateSIPParticipantRequest(ctx context.Context, req *livek
 	}
 	return rpc.NewCreateSIPParticipantRequest(projectID, callID, host, wsUrl, token, req, trunk)
 }
+
+func (s *SIPService) TransferSIPParticipant(ctx context.Context, req *livekit.TransferSIPParticipantRequest) (*emptypb.Empty, error) {
+	log := logger.GetLogger().WithValues("roomName", req.RoomName, "participantIdentity", req.ParticipantIdentity)
+	ireq, err := s.TransferSIPParticipantRequest(ctx, req)
+	if err != nil {
+		log.Errorw("cannot create transfer sip participant request", err)
+		return nil, err
+	}
+
+	timeout := 30 * time.Second
+	if deadline, ok := ctx.Deadline(); ok {
+		timeout = time.Until(deadline)
+	} else {
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+	_, err = s.psrpcClient.TransferSIPParticipant(ctx, ireq.SipCallId, ireq, psrpc.WithRequestTimeout(timeout))
+	if err != nil {
+		log.Errorw("cannot transfer sip participant", err)
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (s *SIPService) TransferSIPParticipantRequest(ctx context.Context, req *livekit.TransferSIPParticipantRequest) (*rpc.InternalTransferSIPParticipantRequest, error) {
```

Comment:
can this be an internal function?

Reply:
I think it's done for Cloud SIP - it will have its own logic for `TransferSIPParticipant`, but the request creation part will be the same.

Reply:
For now, the cloud implementation is likely to be the same as the OSS for this RPC call (we may switch to director eventually). I'll change to not exporting this method for now.